Make in `python/` fails on OS X

I'm getting this error when I run `cd python; make` on a clean checkout of master on OS X Version 10.14.6:

```
In file included from bindings.cc:11:
../tiny_obj_loader.h:312:3: error: unsupported: typedef changes linkage of anonymous type, but linkage was already computed
} material_t;
  ^
../tiny_obj_loader.h:169:15: note: use a tag name here to establish linkage prior to definition
typedef struct {
              ^
               material_t
```

I can't find a clear explanation of what the problem is, though I found a PR which solved a similar issue at falkTX/Carla#652 and the same fix worked. I'm not sure if the fix will make sense to you, or if there is a better way to fix the compilation. I'll open a PR.

```
$ clang --version
Apple clang version 11.0.0 (clang-1100.0.33.8)
Target: x86_64-apple-darwin18.7.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
```

Here's the full build log:

<details>

```
python setup.py build
running build
running build_ext
building 'tinyobjloader' extension
clang -fno-strict-aliasing -fno-common -dynamic -I/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk/usr/include -I/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk/System/Library/Frameworks/Tk.framework/Versions/8.5/Headers -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -I../ -I../pybind11/include -I/usr/local/include -I/usr/local/opt/openssl/include -I/usr/local/opt/sqlite/include -I/usr/local/Cellar/python@2/2.7.16/Frameworks/Python.framework/Versions/2.7/include/python2.7 -c bindings.cc -o build/temp.macosx-10.14-x86_64-2.7/bindings.o -std=c++11
In file included from bindings.cc:1:
In file included from ../pybind11/include/pybind11/pybind11.h:44:
In file included from ../pybind11/include/pybind11/attr.h:13:
In file included from ../pybind11/include/pybind11/cast.h:13:
In file included from ../pybind11/include/pybind11/pytypes.h:12:
In file included from ../pybind11/include/pybind11/detail/common.h:112:
In file included from /usr/local/Cellar/python@2/2.7.16/Frameworks/Python.framework/Versions/2.7/include/python2.7/Python.h:88:
/usr/local/Cellar/python@2/2.7.16/Frameworks/Python.framework/Versions/2.7/include/python2.7/unicodeobject.h:534:5: warning:
      'register' storage class specifier is deprecated and incompatible with C++17 [-Wdeprecated-register]
    register PyObject *obj,     /* Object */
    ^~~~~~~~~
/usr/local/Cellar/python@2/2.7.16/Frameworks/Python.framework/Versions/2.7/include/python2.7/unicodeobject.h:553:5: warning:
      'register' storage class specifier is deprecated and incompatible with C++17 [-Wdeprecated-register]
    register PyObject *obj      /* Object */
    ^~~~~~~~~
/usr/local/Cellar/python@2/2.7.16/Frameworks/Python.framework/Versions/2.7/include/python2.7/unicodeobject.h:575:5: warning:
      'register' storage class specifier is deprecated and incompatible with C++17 [-Wdeprecated-register]
    register const wchar_t *w,  /* wchar_t buffer */
    ^~~~~~~~~
/usr/local/Cellar/python@2/2.7.16/Frameworks/Python.framework/Versions/2.7/include/python2.7/unicodeobject.h:593:5: warning:
      'register' storage class specifier is deprecated and incompatible with C++17 [-Wdeprecated-register]
    register wchar_t *w,        /* wchar_t buffer */
    ^~~~~~~~~
In file included from bindings.cc:1:
In file included from ../pybind11/include/pybind11/pybind11.h:44:
In file included from ../pybind11/include/pybind11/attr.h:13:
In file included from ../pybind11/include/pybind11/cast.h:13:
In file included from ../pybind11/include/pybind11/pytypes.h:12:
In file included from ../pybind11/include/pybind11/detail/common.h:112:
In file included from /usr/local/Cellar/python@2/2.7.16/Frameworks/Python.framework/Versions/2.7/include/python2.7/Python.h:97:
/usr/local/Cellar/python@2/2.7.16/Frameworks/Python.framework/Versions/2.7/include/python2.7/stringobject.h:173:5: warning:
      'register' storage class specifier is deprecated and incompatible with C++17 [-Wdeprecated-register]
    register PyObject *obj,     /* string or Unicode object */
    ^~~~~~~~~
/usr/local/Cellar/python@2/2.7.16/Frameworks/Python.framework/Versions/2.7/include/python2.7/stringobject.h:174:5: warning:
      'register' storage class specifier is deprecated and incompatible with C++17 [-Wdeprecated-register]
    register char **s,          /* pointer to buffer variable */
    ^~~~~~~~~
/usr/local/Cellar/python@2/2.7.16/Frameworks/Python.framework/Versions/2.7/include/python2.7/stringobject.h:175:5: warning:
      'register' storage class specifier is deprecated and incompatible with C++17 [-Wdeprecated-register]
    register Py_ssize_t *len    /* pointer to length variable or NULL
    ^~~~~~~~~
In file included from bindings.cc:11:
../tiny_obj_loader.h:312:3: error: unsupported: typedef changes linkage of anonymous type, but linkage was already computed
} material_t;
  ^
../tiny_obj_loader.h:169:15: note: use a tag name here to establish linkage prior to definition
typedef struct {
              ^
               material_t
s7 warnings and 1 error generated.
error: command 'clang' failed with exit status 1
make: *** [all] Error 1
```

</details>

--- a/tiny_obj_loader.h
+++ b/tiny_obj_loader.h
@@ -166,7 +166,7 @@ typedef struct {
                            // value. Usually `sRGB` or `linear` (default empty).
 } texture_option_t;
 
-typedef struct {
+typedef struct _material_t {
   std::string name;
 
   real_t ambient[3];

